firedrake/mesh.py

```diff
@@ -224,6 +229,41 @@ def facet_cell_map(self):
                        "facet_to_cell_map")
 
 
+@PETSc.Log.EventDecorator()
+def _from_netgen(ngmesh, comm=None):
+    """
+    Create a DMPlex from an Netgen mesh
+
+    :arg ngmesh: NetGen Mesh
+    """
+    if ngmesh.dim == 3:
+        V = ngmesh.Coordinates()
+        T = ngmesh.Indices3D
+        plex = _from_cell_list(3, T, V, comm, name=None)
+        plex.setName(_generate_default_mesh_topology_name(ngmesh.GetMeshName()))
+        for e in ngmesh.Elements2D():
+            vStart, vEnd = plex.getDepthStratum(0)   # vertices
+            join = plex.getFullJoin([vStart+v.nr-1 for v in e.vertices])
+            plex.setLabelValue(dmcommon.FACE_SETS_LABEL, join[0], int(e.index))
+        for e in ngmesh.Elements1D():
+            vStart, vEnd = plex.getDepthStratum(0)   # vertices
+            join = plex.getJoin([vStart+v.nr-1 for v in e.vertices])
+            plex.setLabelValue(dmcommon.FACE_SETS_LABEL, join[0], int(e.index))
```

Comment:
I think this can be `dmcommon.EDGE_SETS_LABEL` after adding the following line at the beginning of `dmcommon.pyx`
```
EDGE_SETS_LABEL = "Edge Sets"
```
"Edge Sets" label does not exist in PETSc, but I don't think it does harm adding it in Firedrake.

Reply:
Done :) Btw If I'd like also to assign a material ID to simplices should I use the CELL_SETS_LABEL?

Reply:
Yes, that's right.